src/Groups/cosets.jl

```diff
@@ -0,0 +1,229 @@
+export right_coset, left_coset, representative, elements, isbicoset, right_cosets, left_cosets, right_transversal, left_transversal, double_coset, acting_domain, left_acting_group, right_acting_group
+
+# T=type of the group, S=type of the element
+"""
+    GroupCoset{T<: Group, S <: GAPGroupElem}
+group coset. It is displayed as `H * x` (right cosets) or `x * H` (left cosets), where `H` is a subgroup of a group `G` and `x` is an element of `G`. Two cosets are equal if, and only if, they are both left (resp. right) and they contain the same elements.
+"""
+mutable struct GroupCoset{T<: Group, S <: GAPGroupElem} 
+   X::T                    # big group containing the subgroup and the element
+   H::T                    # subgroup
+   repr::S                 # element
+   side::Symbol            # says if the coset is left or right
+   coset::GapObj           # GapObj(H*repr)
+end
+
+function _group_coset(X::Group, H::Group, repr::GAPGroupElem, side::Symbol, coset::GapObj)
+  return GroupCoset{typeof(X), typeof(repr)}(X, H, repr, side, coset)
+end
+
+function ==(x::GroupCoset, y::GroupCoset)
+   return x.coset == y.coset && x.side == y.side
+end
+
+"""
+    right_coset(H::Group, g::GAPGroupElem)
+return the coset `Hg`.
+"""
+function right_coset(H::Group, g::GAPGroupElem)
+   @assert elem_type(H) == typeof(g)
+   if !GAP.Globals.IsSubset(parent(g).X, H.X)
+      throw(ArgumentError("H is not a subgroup of parent(g)"))
+   end
+   return _group_coset(parent(g), H, g, :right, GAP.Globals.RightCoset(H.X,g.X))
+end
+
+"""
+    right_coset(H::Group, g::GAPGroupElem)
+return the coset `gH`.
+"""
+function left_coset(H::Group, g::GAPGroupElem)
+   @assert elem_type(H) == typeof(g)
+   if !GAP.Globals.IsSubset(parent(g).X, H.X)
+      throw(ArgumentError("H is not a subgroup of parent(g)"))
+   end
+   return _group_coset(parent(g), H, g, :left, GAP.Globals.RightCoset(GAP.Globals.ConjugateSubgroup(H.X,GAP.Globals.Inverse(g.X)),g.X))
+end
+
+function show(io::IO, x::GroupCoset)
+   if x.side == :right
```

Comment:
```julia
a = GAP.gap_to_julia(GAP.Globals.StringView(x.H.X))
b = GAP.gap_to_julia(GAP.Globals.StringView(x.repr.X))
if x.side == :right
    print(io, a, " * ", b)
else
    print(io, b, " * ", a)
end
```
is much more readable (at least to my eyes); You may also think about not using `x.side` directly, but to define `side` function for future maintainability